developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+- Easier Onboarding & Adoption - When dev essential tasks are automated, there is less documentation to write and, subsequently, less to read to get started!
+
+## Measures
+
+### Time to First Debug (aka F5 Contract)
+
+Assuming a laptop/pc that has never run the solution, how long does it take to start debugging the solution end-to-end.
+
+### Time To First Commit
+
+How long does it take to make a change that can be verified/tested locally. A locally verified/tested change is one that passes test cases without introducing regression or breaking changes.
+
+## Participation
+
+The entire development team should take responsibility in ensuring positive
+
+## Facilitation Guidance
+
+The following outline several strategies that can be adopted to promote a positive developer experience.
+
+### Establish Hotkeys
+
+Assign hotkeys to each of the essential tasks.
+
+[`TODO: Suggest hotkeys for mac`]
+
+| Task                 | Windows      |
+| -------------------- | ------------ |
+| Build                | CTRL+SHIFT+B |
+| Test                 | CTRL+R,T     |
+| Start With Debugging | F5           |
+
+### The F5 Contract
+
+The F5 contract aims for the ability to run the end-to-end solution with the following steps.
+
+1. Clone - git clone [`my-repo-url-here`]
+2. Configure - set any configuration values that need to be unique to the individual (i.e. update a .env file)
+3. Press F5 - launch the solution with debugging attached.
+
+Most IDE's have some form of a task runner that can be used to automate the build, execute, and attach steps. Try to leverage these such that the steps can all be run with as few manual steps as possible.
+
+### Pay Attention to the Struggles
+
+Ask the team to surface the areas they spend significant time in that they consider non-value-add activities. For example, do they need to deploy their changes to an environment off their laptop before they can validate if what they did worked. Rather than debugging locally, do they have to do this repetitively to get to a working solution? Does this take several minutes each iteration? Does this block other developers due to the contention on the environment?
+
+The following are a few strategies for surfacing dev experience struggles
+
+- Retrospectives. Create a card that describes the struggle
+- Standup Blockers. Call out being blocked not being able to test change and why that cant be done.
+
+#### Fixing the Struggles
+
+Once raised, use the following steps to capture and create **actionable** fix.
+
+1. Define the problem - End to end testing is only possible when the solution is deployed to environment in Azure
+2. Identify a potential solution - Ability to run automated end-to-end tests on the developer machine.
+3. Justify fixing the problem - The team has identified this issue as the root cause for 3 stories not completing within their sprint over the past 2 sprints.
+
+### Single IDE Instance

```

Comment:
Not sure what the guidance would be for somebody who prefers to develop purely in a terminal, but is it possible to generalize this section to focus on the core problem being discussed here without assuming a specific technology?

Here are some thoughts on how this may be more generally written:
 - Standardizing the configuration for solution components
 - Standardizing the way tests are run for each component
 - Standardizing on a small number of technology stacks
 - Standardizing on documentation styles and practices

Reply:
I think this is closely related to the single repo section so maybe it should be a subsection within there. However, the pain i have felt from past projects that have a lots of repo's is you end up having to manage multiple instances of VSCode, Visual Studio, Sublime Text or whatever your preferred IDE is. Also this almost always multiplies the number of steps it takes to configure, build, test, and start the entire solution. 

This issue also leads teams to start to favor horizontal sliced work as a means to mitigate all the inefficiencies with managing all the repo's and IDE's. This starts to create even more issues down the road.

So i think what im trying to suggest is that being to develop in a single IDE is a strategy to prevent this from happening. Does that clarification change what your suggestion?

Reply:
I think that the "multiple IDE" problem you are describing is the solution to the underlying issue that you described well in your comment:
> Also this almost always multiplies the number of steps it takes to configure, build, test, and start the entire solution. ... This issue also leads teams to start to favor horizontal sliced work as a means to mitigate all the inefficiencies with managing all the repo's and IDE's. This starts to create even more issues down the road.

My fear is that somebody takes the "single IDE" suggestion literally but does not fix the underlying issue. An example of this is my current project, where I use 1 IDE instance to open 10 git repositories at any given time. The challenge is still there, even though I only need 1 IDE instance for my work.

Reply:
@nmiodice I replaced this section with two others that combine your very valid point about standardization of essential across components and automating those tasks across the solution. I think this takes an approach at describing a potential solution that is not necessarily tied to an IDE. For example, I think that someone could easily see implementing a solution level build with a shell script would be fine (in the case they are hard-core enough to be using notepad/sublime + terminal). Let me know what you think of this update and if it better focuses on the root problem/solution.